testing/execution.ts

```diff
@@ -924,3 +933,30 @@ function parseGetPermissionRequest(
 
   return {permissionRequest: parseResult.data, params: coerceParams(syncFormula, paramsCopy as any)};
 }
+
+function isSourceMapsEnabled() {
+  const flags = [
+    ...process.execArgv,
+    ...process.env.NODE_OPTIONS?.split(/\s+/) ?? [],
+  ];
+  return flags.includes('--enable-source-maps');
+}
+
+class DeveloperError extends Error {
+  constructor(err: Error) {
+    super('The Pack code threw an error: ' + err.message, {
+      cause: err,
+    });
+    this.stack = err.stack;
+    this.name = 'DeveloperError';
+    Object.setPrototypeOf(this, DeveloperError.prototype);
+  }
+}
+
+function asDeveloperError(err: Error) {
+  throw new DeveloperError(err);
+}
+
+function isDeveloperError(err: Error) {
+  return err.name === 'DeveloperError';
```

Comment:
Can you do `err instanceof DeveloperError`? By convention that's how to check this, though internally I believe `instanceof` is actually just checking the names.

Reply:
Yep, that's much simpler! That, and the helper functions, were a vestige from a previous approach using a Proxy class. Cleaned up now.